server/src/test/java/org/opensearch/index/shard/RemoteStoreRefreshListenerTests.java

```diff
@@ -254,7 +270,39 @@ public void testRefreshSuccessOnThirdAttemptAttempt() throws Exception {
         assertBusy(() -> assertEquals(0, successLatch.getCount()));
     }
 
-    private void mockIndexShardWithRetryAndScheduleRefresh(
+    public void testTrackerData() throws Exception {
+        Tuple<RemoteStoreRefreshListener, RemoteRefreshSegmentPressureService> tuple = mockIndexShardWithRetryAndScheduleRefresh(1);
+        RemoteStoreRefreshListener listener = tuple.v1();
+        RemoteRefreshSegmentPressureService pressureService = tuple.v2();
+        RemoteRefreshSegmentTracker tracker = pressureService.getRemoteRefreshSegmentTracker(indexShard.shardId());
+        assertNoLag(tracker);
+        indexDocs(100, randomIntBetween(100, 200));
+        indexShard.refresh("test");
+        listener.afterRefresh(true);
+        assertBusy(() -> assertNoLag(tracker));
+    }
+
+    private void assertNoLag(RemoteRefreshSegmentTracker tracker) {
+        assertEquals(0, tracker.getRefreshSeqNoLag());
+        assertEquals(0, tracker.getBytesLag());
+        assertEquals(0, tracker.getTimeMsLag());
+        assertEquals(0, tracker.getRejectionCount());
+        assertEquals(tracker.getUploadBytesStarted(), tracker.getUploadBytesSucceeded());
+        assertTrue(tracker.getUploadBytesStarted() > 0);
+        assertEquals(0, tracker.getUploadBytesFailed());
+        assertEquals(0, tracker.getInflightUploads());
+        assertEquals(tracker.getTotalUploadsStarted(), tracker.getTotalUploadsSucceeded());
+        assertTrue(tracker.getTotalUploadsStarted() > 0);
+        assertEquals(0, tracker.getTotalUploadsFailed());
```

Comment:
can we assert count of failed uploads in tests which have failures and retries covered already ?

Reply:
done